applications/GeoMechanicsApplication/custom_processes/set_parameter_field_process.cpp

```diff
@@ -28,132 +28,132 @@ namespace Kratos
+        const double current_time = this->mrModelPart.GetProcessInfo().GetValue(TIME);
+        const auto indices = GetVectorIndices();
+
+        for (Element& r_element : mrModelPart.Elements()) {
+            const auto& r_geom = r_element.GetGeometry();
+
+            // calculate parameter value at current element
+            const double val = parameter_function.CallFunction(r_geom.Center().X(), r_geom.Center().Y(), r_geom.Center().Z(), current_time, 0, 0, 0);
+            // get properties per element
+            Properties& r_prop = r_element.GetProperties();
+            Vector& vector = r_prop.GetValue(rVar);
+
+            // loop through the indexes that need to be set
+            for (auto index : indices) {
+                vector[index] = val;
+            }
+
+            SetValueAtElement(r_element, rVar, vector);
+        }
     }
-}
-
-void SetParameterFieldProcess::SetParameterFieldUsingJsonString(const  Variable<double>& rVar)
-{
-    // get new data from the data set
-    const std::string& r_dataset = mParameters["dataset"].GetString();
-
-
-    Parameters new_data{ r_dataset };
-    this->SetParameterFieldUsingParametersClass(rVar, new_data);
 
-}
-void SetParameterFieldProcess::SetParameterFieldUsingJsonFile(const Variable<double>& rVar)
-{
-    // Read json string in field parameters file, create Parameters
-    const std::string& field_file_name = mParameters["dataset_file_name"].GetString();
-    KRATOS_ERROR_IF_NOT(std::filesystem::exists(field_file_name)) << "The parameter field file specified with name \"" << field_file_name << "\" does not exist!" << std::endl;
+    void SetParameterFieldProcess::SetParameterFieldUsingParametersClass(const Variable<double>& rVar,
+        const Parameters& rParameters)
+    {
+        const Vector& r_data_vector = rParameters["values"].GetVector();
 
-    std::ifstream ifs(field_file_name);
-    Parameters new_data{ ifs };
+        KRATOS_ERROR_IF_NOT(r_data_vector.size() == mrModelPart.Elements().size()) << "The parameter field "
+            "does not have the same size as the amount of elements within the model part!" << std::endl;
 
-    this->SetParameterFieldUsingParametersClass(rVar, new_data);
+        // set new data on the elements
+        IndexType i = 0;
+        for (Element& r_element : mrModelPart.Elements()) {
+            SetValueAtElement(r_element, rVar, r_data_vector[i]);
+            ++i;
+        }
+    }
 
-}
+    void SetParameterFieldProcess::SetParameterFieldUsingParametersClass(const Variable<Vector>& rVar,
+        const Parameters& rParameters)
+    {
+        const Matrix& r_data_matrix = rParameters["values"].GetMatrix();
+
+        KRATOS_ERROR_IF_NOT(r_data_matrix.size1() == mrModelPart.Elements().size()) << "The parameter field "
+            "does not have the same size as the amount of elements within the model part!" << std::endl;
+
+        // set new data on the elements
+        const int vector_size = r_data_matrix.size2();
```

Comment:
Please note that SonarQube reports a narrowing conversion here (i.e. an implicit conversion through which we loose precision). I would suggest you to use the `size_type` of the matrix class rather than narrowing it to an `int`.

Reply:
Ok!

Reply:
Changing the type of the vector_size to IndexType which is the size_type of the matrix class make some ubuntu builders fail. So I will revert it back to the int since then the tests pass. Any other suggestions are welcome